### DISCUSSION TITLE
SpringBootTest의 WebEnvironment 설정 종류를 설명해주세요.

### DISCUSSION THREAD

**Discussion by Kimprodp** [2024-woowacourse-study/level-interview#141] [Spring Test]
.

**Comment by Kimprodp** [comment]
SpringBootTest 어노테이션의 webEnvironment 속성은 Spring Boot 테스트에서 웹 환경을 설정하는 데 사용된다.

**webEnvironment = SpringBootTest.WebEnvironment.NONE**
- 웹 환경을 설정하지 않음
- 웹 계층을 포함하지 않는 서비스나 리포지토리 테스트에 적합

**webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT**
- 실제 웹 서버를 기본 포트에서 시작(application.properties 정의된 포트)
- 특정 포트에서 애플리케이션을 테스트해야 할 때 유용

**webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT**
- 실제 웹 서버를 랜덤한 포트에서 시작
- 애플리케이션이 네트워크 상에서 실제로 동작하는지 확인할 때 유용
- 테스트 중 포트 충돌을 피할 수 있음

**webEnvironment = SpringBootTest.WebEnvironment.MOCK**
- 가짜 서블릿 환경을 제공(실제 웹 서버를 시작하지 않고, Spring MVC 테스트 프레임워크를 사용하여 서블릿 환경을 모킹)
- 주로 컨트롤러 테스트에서 사용되며, 실제 네트워크 요청을 보내지 않고도 웹 계층을 테스트할 수 있음
- webEnvironment 설정을 하지 않으면 디폴트로 적용됨

**Comment by seunghye218** [comment]
## WebEnvironment 속성

@SpringBootTest의 webEnvironment 속성은 테스트 실행 시 애플리케이션의 웹 환경을 설정하는 데 사용됩니다. 이 속성을 통해 테스트에서 사용할 웹 환경의 종류를 지정할 수 있습니다. Spring Boot는 네 가지 주요 웹 환경을 제공합니다. 각 설정은 테스트의 목적과 필요에 따라 선택할 수 있습니다.

### MOCK

모킹된 서블릿 환경을 제공합니다. 실제 웹 서버를 시작하지 않고, 내장된 모킹된 서블릿 환경을 사용합니다.
컨트롤러 테스트와 같은 빠른 단위 테스트를 수행할 때 사용됩니다. 외부 서버가 필요 없고, 빠르게 테스트를 실행할 수 있습니다.

### RANDOM_PORT

실제 내장 웹 서버를 시작하고, 랜덤 포트에서 애플리케이션을 실행합니다. 테스트 실행 시 매번 다른 포트를 사용하여 충돌을 피합니다.
실제 서버 환경에서 통합 테스트를 수행할 때 사용됩니다. 예를 들어, REST API 테스트에 적합합니다.

### DEFINED_PORT

실제 내장 웹 서버를 시작하고, 애플리케이션이 정의된 기본 포트(기본값 8080)에서 실행됩니다. 애플리케이션이 항상 동일한 포트에서 실행됩니다. application.properties 또는 application.yml 파일에서 포트를 설정할 수 있습니다. 특정 포트에서 실행되는 서버를 필요로 하는 통합 테스트에 사용됩니다.

### NONE

웹 환경을 비활성화합니다. 내장 웹 서버를 시작하지 않습니다. 순수한 스프링 컨텍스트 테스트 또는 웹 서버가 필요 없는 서비스 레이어 테스트에 사용됩니다.
웹 환경이 필요 없는 테스트, 예를 들어 서비스나 리포지토리 테스트에 적합합니다.
